docs/APIs/storage/block_device.md

```diff
@@ -1,38 +1,36 @@
 # Block Devices
 
-The block device API provides a simple interface for providing access to block-based storage. A block device can be used to back a full [filesystem](filesystem.md) or can be written to
-directly.
+The block device API provides an interface for access to block-based storage. You can use a block device to back a full [file system](filesystem.md) or write to it directly.
 
-The full C++ API can be found [here](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/bd/BlockDevice.h)
+You can find the full C++ API [here](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/bd/BlockDevice.h).
 
 ## Example
 
-Here is a quick example of a simple user of a block device:
+Here is an example that uses a block device:
 
 [SD block device example](https://github.com/ARMmbed/sd-driver/blob/master/README.md)
 
 ## Block device operations
 
-A block device can perform three operations on block in a device:
+A block device can perform three operations on a block in a device:
 
-- Read a block from storage
-- Erase a block in storage
-- Program a block that has previously been erased
+- Read a block from storage.
+- Erase a block in storage.
+- Program a block that has previously been erased.
 
-Note: The state of an erased block is undefined. For some block devices, erase is a noop. If a deterministic value is required after an erase, this must be verified by the consumer of the block device.
```

Comment:
"The state of an erased block is undefined"

I think it would be good to elaborate the comment here i.e. the erased block will have specific define values for some block devices e.g. NOR flash parts typically have an specific value for a byte that's been erased e.g. 0xff. Drivers reporting ARM_FLASH_INFO.erased_value or ARM_STORAGE_INFO.erased_value use this attribute to report the value of an erased block to driver clients.